deltacat/compute/compactor/steps/materialize.py

```diff
@@ -45,16 +47,20 @@ def materialize(
         compacted_file_content_type: ContentType,
         schema: Optional[pa.Schema] = None,
         deltacat_storage=unimplemented_deltacat_storage) -> MaterializeResult:
-
     def _materialize(
-            compacted_tables: List[pa.Table],
-            compacted_tables_record_count: int) -> MaterializeResult:
+            compacted_tables: List[pa.Table]) -> MaterializeResult:
         compacted_tables_size = sum([TABLE_CLASS_TO_SIZE_FUNC[type(tbl)](tbl)
                                      for tbl in compacted_tables])
-        logger.debug(f"Uploading {len(compacted_tables)} compacted tables "
-                     f"with size: {compacted_tables_size} bytes "
-                     f"and record count: {compacted_tables_record_count}")
+        logger.debug(f"Concatenating {len(compacted_tables)} compacted tables"
+                     f" with total size: {compacted_tables_size} bytes")
         compacted_table = pa.concat_tables(compacted_tables)
+        compacted_table = compacted_table.take(
+            pc.sort_indices(compacted_table, sort_keys=PA_SORT_KEYS),
+        )
+        compacted_table = compacted_table.drop(
+            [sc._DEDUPE_TASK_IDX_COLUMN_NAME]
+        )
+
```

Comment:
To be precise, this sorting can be removed as we are not predicting the indices during dedupe.